### PR TITLE
Add basic reporter

### LINKED ISSUE #53: Report errors in a JSON file
To assist in debugging, we should log out errors

### PR DESCRIPTION
**Pull request by qgolsteyn**
Moves parsing and writing data into its own files. Creates a basic reporting object which is written when processing is completed.

Closes #53

### CODE REVIEW

File: tasks/findFeatures.js [hunk 1]
```diff
@@ -273,14 +274,24 @@ function generateFeatures({ locations }) {
       }
 
       if (!found) {
+<<<<<<< HEAD
+        console.error('  ❌ Could not find location %s [%f, %f]', transform.getName(location), location.coordinates ? location.coordinates[0] : '?', location.coordinates ? location.coordinates[1] : '?');
+        errors.push(transform.getName(location));
+=======
```
Comment (lazd) [inline tasks/findFeatures.js]: Little merge conflict here

File: tasks/findFeatures.js [hunk 2]
```diff
@@ -273,14 +274,24 @@ function generateFeatures({ locations }) {
...
     }
 
     console.log('✅ Found features for %d out of %d regions for a total of %d features', foundCount, Object.keys(locations).length, featureCollection.features.length);
 
+    report['findFeatures'] = {
+      numFeaturesFound: foundCount,
+      missingFeatures: errors
+    };
+
     resolve({ locations, featureCollection });
```
Comment (lazd) [inline tasks/findFeatures.js]: I'd say pass the report downstream:

```
resolve({ locations, featureCollection, report });
```

**Comment by lazd** [comment]
I like the modularity. I say pass the report downstream as part of the object and it's good.

**Comment by qgolsteyn** [comment]
@lazd Addressed code review

**Comment by lazd** [comment]
Ok, so it's both building the object, and printing the errors for everything, right?

This looks amazing.

**Pull request merged**